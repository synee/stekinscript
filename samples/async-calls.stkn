func asyncTextGen(input, cb)
    if true
        setTimeout(%, 1)
        if input.length < 4
            cb(input + ' is quite short')
        else
            cb(input + ' is quite long')
    return input.length

console.log(asyncTextGen('cat', %text))
console.log(text)
console.log(asyncTextGen(text, %moreText))
console.log(moreText)

regularAsyncTextGen: (input, %%):
    setTimeout(%, 1)
    return input + (' is quite short' if input.length < 4 else ' is quite long')

console.log(regularAsyncTextGen('cat', %%))
console.log(regularAsyncTextGen('raven', %%))

func cancelAtFirstTime(timeoutHandle)
    clearTimeout(timeoutHandle)

cancelAtFirstTime(setInterval(%, 1))

func asyncRaiseError(callback)
    setTimeout(%, 1)
    callback('error message')

((%%):
    asyncRaiseError(%%)
)((error, result):
    console.log(error)
)

console.log('End')
