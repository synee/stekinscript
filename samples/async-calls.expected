3
cat is quite short
18
cat is quite short is quite long
cat is quite short
raven is quite long
End
error message
